src/main/java/uk/gov/companieshouse/psc/extensions/api/controller/HealthCheckController.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.companieshouse.psc.extensions.api.controller;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HealthCheckController {
+
+  @GetMapping("/psc-extensions-api/healthcheck")
+  public Map<String, String> healthCheck() {
+    Map<String, String> response = new HashMap<>();
+    response.put("status", "UP");
+    return response;
```

Comment:
This can be replaced with the following 

```suggestion
    return Map.of("status", "UP");
```

Not only is this less lines but creating a full `HashMap` means that 16 buckets are allocated which means there's wasted memory if you're not going to use them.

`Map.of` will create an immutable map of the exact size you need.

Reply:
this controller is going to be removed due to previous comments